### PR TITLE
For diagnostic information of Boolean, remind it as use the type: 'bool'

### LINKED ISSUE #98492: Provide a pointer to the correct names for Boolean types
Given the following code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4e7d539241dfe92a24817728751e5523

```rust
fn main() {
    let value: Boolean = true;
}
```

The current output is:

```
error[E0412]: cannot find type `Boolean` in this scope
 --> src/main.rs:2:16
  |
2 |     let value: Boolean = true;
  |                ^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0412`.
error: could not compile `example` due to previous error
```

Ideally the output should look like:

```
error[E0412]: cannot find type `Boolean ` in this scope
    --> src/main.rs:4:16
     |
4    |       let value: Boolean = true;
     |                  ^^^^^^^ help: a type with a similar meaning exists: `bool`
```

I stumbled here when returning to Rust after a while writing code in non-Rust languages; I'm sure I'm not the only person who's gotten the name of Rust's boolean type wrong. It's not hard to work out the correct behaviour, but I thought it would be nice if Rust could help programmers coming from other languages.

(I don't feel strongly about this suggestion and won't mind if it's closed as WONTFIX, but I know Esteban would be sad if he knew I'd had an idea for a diagnostic improvement that I didn't tell anyone.)

### PR DESCRIPTION
**Pull request by lyming2007**
Fixes #98492.

It helps programmers coming from other languages
	modified:   compiler/rustc_resolve/src/late/diagnostics.rs

### CODE REVIEW

File: src/test/ui/typeck/issue-98492.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+fn main() {
+  let value: Boolean = true;
```
Comment (lcnr) [inline src/test/ui/typeck/issue-98492.rs:2]: can you instead add this to `src/test/ui/lint/recommended-literal.rs`?

File: compiler/rustc_resolve/src/late/diagnostics.rs [hunk 2]
```diff
@@ -1508,6 +1508,7 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         Some(match name {
             "byte" => sym::u8, // In Java, bytes are signed, but in practice one almost always wants unsigned bytes.
             "short" => sym::i16,
+            "Boolean" => sym::bool,
```
Comment (camelid) [inline compiler/rustc_resolve/src/late/diagnostics.rs:1511]: Perhaps would make sense to add "Bool" to this list too? It's common in functional languages like Haskell.

```suggestion
            "Bool" => sym::bool,
            "Boolean" => sym::bool,
```

File: src/test/ui/lint/recommend-literal.rs [hunk 3]
```diff
@@ -7,6 +7,8 @@ fn main() {
     let y: long = 74802374902374923;
     //~^ ERROR cannot find type `long` in this scope
     //~| HELP perhaps you intended to use this type
+    let value: Boolean = true;
```
Comment (camelid) [inline src/test/ui/lint/recommend-literal.rs:10]: It'd be good to add a test for `Bool` as well.

**Pull request merged**